var/spack/repos/builtin/packages/hipsycl/package.py

```diff
@@ -0,0 +1,114 @@
+
+    maintainers = ["nazavode"]
+
+    provides("sycl")
+
+    version("master", branch="master", submodules=True)
+    version(
+        "0.8.0",
+        commit="2daf8407e49dd32ebd1c266e8e944e390d28b22a",
+        submodules=True,
+    )
+
+    variant(
+        "cuda",
+        default=False,
+        description="Enable CUDA backend for SYCL kernels",
+    )
+
+    depends_on("cmake@3.5:", type="build")
+    depends_on("boost +filesystem")
+    depends_on("python@3:")
+    depends_on("llvm@8: +clang", when="~cuda")
+    depends_on("llvm@9: +clang", when="+cuda")
+    # LLVM PTX backend requires cuda7:10.1.9999 (https://tinyurl.com/v82k5qq)
+    depends_on("cuda@9:10.1.9999", when="@0.8.1: +cuda")
+    # hipSYCL@:0.8.0 requires cuda@9:10.0.9999 due to a known bug
+    depends_on("cuda@9:10.0.9999", when="@:0.8.0 +cuda")
+
+    conflicts(
+        "^llvm build_type=Debug",
+        when="+cuda",
+        msg="LLVM debug builds don't work with hipSYCL CUDA backend; for "
+        "further info please refer to: "
+        "https://github.com/illuhad/hipSYCL/blob/master/doc/install-cuda.md",
+    )
+
+    def cmake_args(self):
+        spec = self.spec
+        args = [
+            "-DWITH_CPU_BACKEND:Bool=TRUE",
+            # TODO: no ROCm stuff available in spack yet
+            "-DWITH_ROCM_BACKEND:Bool=FALSE",
+            "-DWITH_CUDA_BACKEND:Bool={0}".format(
+                "TRUE" if "+cuda" in spec else "FALSE"
+            ),
+            # prevent hipSYCL's cmake to look for other LLVM installations
+            # if the specified one isn't compatible
+            "-DDISABLE_LLVM_VERSION_CHECK:Bool=TRUE",
+        ]
+        # LLVM directory containing all installed CMake files
+        # (e.g.: configs consumed by client projects)
+        llvm_cmake_dirs = filesystem.find(
+            spec["llvm"].prefix, "LLVMExports.cmake"
+        )
+        if len(llvm_cmake_dirs) != 1:
+            raise InstallError(
+                "concretized llvm dependency must provide "
+                "a unique directory containing CMake client "
+                "files, found: {0}".format(llvm_cmake_dirs)
+            )
+        args.append(
+            "-DLLVM_DIR:String={0}".format(path.dirname(llvm_cmake_dirs[0]))
+        )
+        # clang internal headers directory
+        llvm_clang_include_dirs = filesystem.find(
+            spec["llvm"].prefix, "__clang_cuda_runtime_wrapper.h"
+        )
+        if len(llvm_clang_include_dirs) != 1:
+            raise InstallError(
+                "concretized llvm dependency must provide a "
+                "unique directory containing clang internal "
+                "headers, found: {0}".format(llvm_clang_include_dirs)
+            )
+        args.append(
+            "-DCLANG_INCLUDE_PATH:String={0}".format(
+                path.dirname(llvm_clang_include_dirs[0])
+            )
+        )
+        # target clang executable
+        llvm_clang_bin = path.join(spec["llvm"].prefix.bin, "clang")
```

Comment:
This should be fixed now, thanks for pointing it out.

Reply:
@illuhad just fixed the issue related to missing `rpath`s to `libc++.so` and `libc++abi.so`. Since I have no way to actually test the `ROCm` backend (`ROCm` stuff not available yet in Spack), the issue is fixed just for  `--hipsycl-platform=cuda` while `--hipsycl-platform=cpu` seems not to be affected (`libc++*` stuff looks to be linked only as transitive link dependencies via `libhipSYCL_cpu.so` which in turns correctly has everything found via `rpath`).

Reply:
Yes, seems fine now, thanks! I don't think we have to worry about ROCm as long as it is not in spack. ROCm is very complex, and the way it has to be linked also depends on how it is built - I think we'll just have to wait and see how the ROCm spack package behaves (assuming it comes at some point in the future)